airstable.rb

```diff
@@ -0,0 +1,256 @@
+  else
+    sign_in!(user)
+    redirect('/dashboard')
+  end
+end
+
+get '/users/new' do
+  user = User.new
+  erb :'/users/new', locals: { user: user, title: 'Create an Account' }
+end
+
+post '/users/new' do
+  user = User.create(params[:user])
+
+  if user.saved?
+    sign_in!(user)
+    redirect('/dashboard')
+  else
+    erb :'/users/new', locals: { user: user, title: 'Create an Account' }
+  end
+end
+
+get '/users/:id/edit' do
+  # Render edit user
+end
+
+post '/users/:id' do
+  # Process edit user
+end
+
+get '/dashboard' do
+  erb :'/dashboard', locals: { user: current_user, title: 'My Dashboard' }
+end
+
+get '/stalls' do
+  stalls = Stall.all
+
+  erb :'/stalls/index', locals: { stalls: stalls, title: 'All Stalls' }
+end
+
+get '/stalls/new' do
+  stall = Stall.new
+
+  erb :'/stalls/new', locals: { stall: stall, title: 'Create a Stall' }
+end
+
+get '/stalls/:id' do
+  stall = Stall.get(params[:id])
+
+  erb :'/stalls/show', locals: { stall: stall, title: "#{stall.title}" }
+end
+
+post '/stalls' do
+  stall = Stall.create(params[:stall])
+
+  if stall.saved?
+    redirect("/stalls/#{stall.id}")
+  else
+    erb :'/stalls/new', locals: { stall: stall, title: 'Create a Stall' }
+  end
+end
+
+get '/stalls/:id/edit' do
+  stall = Stall.get(params[:id])
+
+  erb :'/stalls/edit', locals: { stall: stall, title: "Edit #{stall.title}" }
+end
+
+put '/stalls/:id' do
+  stall = Stall.get(params[:id])
+  stall.attributes = params[:stall] if params[:stall]
+  stall.save
+
+  puts "Stall dirty?: #{stall.dirty?}"
+  if stall.saved?
+    puts "Stall saved?: #{stall.saved?}"
+    redirect("/stalls/#{stall.id}")
+  else
+    erb :'/stalls/new', locals: { stall: stall }
+  end
+end
+
+delete '/stalls/:id' do
+  # delete a stall (cancel any pending or accepted requests)
+end
+
+get '/stalls/:id/rental_requests/new' do
+  stall = Stall.get(params[:id])
+  rental_request = RentalRequest.new
+
+  erb :'/stalls/rental_requests/new',
+      locals: {
+        title: "Request #{stall.title}",
+        stall: stall,
+        rental_request: rental_request
+              }
+end
+
+post '/stalls/:id/rental_requests' do
+  PP.pp params
+  stall = Stall.get(params[:id])
+
+  rental_request = RentalRequest.new(params[:rental_request])
+  rental_request.status = :pending
+  rental_request.stall = stall
+  rental_request.requester = current_user
+  rental_request.save
+
+  if rental_request.saved?
+    set_flash(:notice, 'Rental request created!')
+    redirect '/dashboard'
+  else
+    PP.pp rental_request
+    PP.pp rental_request.errors
+    erb :"/stalls/rental_requests/new",
+        locals: {
+          title: "Request #{stall.title}",
+          stall: stall,
+          rental_request: rental_request
+                }
+  end
+end
+
+put '/stalls/:stall_id/rental_requests/:rental_request_id/edit' do
+  rental_request = RentalRequest.get(params[:rental_request_id])
+  rental_request.status = params['rental_request_response'].to_sym
```

Comment:
Is it necessary to convert this to a `Symbol` before saving?  I'd be surprised if DataMapper didn't do that internally.